File: TesminePoulose_COMP304_Sec001_Lab3/app/src/main/java/com/example/tesminepoulose_comp304_sec001_lab3/CanvasActivity.java
package com.example.tesminepoulose_comp304_sec001_lab3;


import android.graphics.Bitmap;
import android.graphics.Canvas;
import android.graphics.Color;
import android.graphics.Paint;
import android.os.Bundle;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.Spinner;
import android.widget.TextView;

import androidx.appcompat.app.AppCompatActivity;

import java.util.ArrayList;
import java.util.List;

public class CanvasActivity extends AppCompatActivity {

    private ImageView reusableImageView;
    private TextView textView;
    //
    private int startx = 0;
    private int starty = 0;
    private int endx=0;
    private int endy=0;
    //
    private Paint paint;
    private Bitmap bitmap;
    private Canvas canvas;
    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_canvas);
        Spinner spinner = findViewById(R.id.spinner1);
        final List<Integer> dataList = new ArrayList<Integer>();
        for(int i=10;i<100;i=i+5)
        {
            dataList.add(i);
        }
        ArrayAdapter<Integer> arrayAdapter = new ArrayAdapter<Integer>(this, android.R.layout.simple_spinner_dropdown_item, dataList);
        spinner.setAdapter(arrayAdapter);



        paint = new Paint();
        paint.setColor(Color.RED);
        //paint.setStrokeWidth(30);

        bitmap = Bitmap.createBitmap(getWindowManager()
                .getDefaultDisplay().getWidth(), getWindowManager()
                .getDefaultDisplay().getHeight(), Bitmap.Config.ARGB_8888);
        canvas = new Canvas(bitmap);
        canvas.drawColor(Color.WHITE); //background
        //
        reusableImageView = findViewById(R.id.ImageViewForDrawing);
        reusableImageView.setImageBitmap(bitmap);
        reusableImageView.setVisibility(View.VISIBLE);


        canvas.drawPoint(1,1,paint);



        spinner.setOnItemSelectedListener(new  AdapterView.OnItemSelectedListener() {

            public void onItemSelected(AdapterView<?> adapterView,
                                       View view, int i, long l) {

                paint.setStrokeWidth(dataList.get(i));

            }
            // If no option selected
            public void onNothingSelected(AdapterView<?> arg0) {
            }

        });

        textView = findViewById(R.id.help);
        textView.setText("Use Arrow Keys to draw ! ");

    }

    public void drawright(View view) {
        reusableImageView.setFocusable(true);
        reusableImageView.requestFocus();
        endx=endx+5;
        canvas.drawLine(startx, starty, endx, endy, paint);
        startx=endx;
        starty=endy;
        //moveRect(canvas);
        reusableImageView.invalidate();
        textView.setText("X= "+endx+"  Y= "+ endy);
    }

    public void drawdown(View view) {
        reusableImageView.setFocusable(true);
        reusableImageView.requestFocus();
        endy=endy+5;
        canvas.drawLine(startx, starty, endx, endy, paint);
        startx=endx;
        starty=endy;
        //moveRect(canvas);
        reusableImageView.invalidate();
        textView.setText("X= "+endx+"  Y= "+ endy);
    }

    public void drawleft(View view) {
        reusableImageView.setFocusable(true);
        reusableImageView.requestFocus();
        endx=endx-5;
        canvas.drawLine(startx, starty, endx, endy, paint);
        startx=endx;
        starty=endy;
        //moveRect(canvas);
        reusableImageView.invalidate();
        textView.setText("X= "+endx+"  Y= "+ endy);
    }

    public void drawup(View view) {
        reusableImageView.setFocusable(true);
        reusableImageView.requestFocus();
        endy=endy-5;
        canvas.drawLine(startx, starty, endx, endy, paint);
        startx=endx;
        starty=endy;
        //moveRect(canvas);
        reusableImageView.invalidate();
        textView.setText("X= "+endx+"  Y= "+ endy);
    }

    public void redline(View view) {

        paint.setColor(Color.RED);
    }

    public void cyan(View view) {
        paint.setColor(Color.CYAN);
    }

    public void yellowline(View view) {
        paint.setColor(Color.YELLOW);
    }

    public void clearall(View view) {
        canvas.drawColor(Color.rgb(95, 75, 92)); //background
        canvas.drawColor(Color.WHITE);
        startx = 0;
        starty = 0;
        endx = 0;
        endy = 0;
        textView.setText("Use Arrow Keys to draw ! ");

    }
}

File: app/src/main/java/com/example/gokulrajvenugopal_comp304lab3/CanvasActivity.kt
package com.example.gokulrajvenugopal_comp304lab3

import android.app.AlertDialog
import android.graphics.*
import android.os.Bundle
import android.view.View
import android.widget.AdapterView
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.Spinner
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity

/*
Student Name: Gokulraj Venugopal
Student ID: 301202722
 */

class CanvasActivity : AppCompatActivity() {
    private lateinit var reusableImageView: ImageView
    private lateinit var textView: TextView

    //
    private var startx = 0
    private var starty = 0
    private var endx = 0
    private var endy = 0

    //
    private lateinit var paint: Paint
    private lateinit var bitmap: Bitmap
    private lateinit var canvas: Canvas

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_canvas)
        val spinner = findViewById<Spinner>(R.id.spinner1)
        val dataList = mutableListOf<Int>()
        for (i in 10 until 100 step 5) {
            dataList.add(i)
        }
        val arrayAdapter =
            ArrayAdapter(this, android.R.layout.simple_spinner_dropdown_item, dataList)
        spinner.adapter = arrayAdapter

        paint = Paint()
        paint.color = Color.RED
        //paint.strokeWidth = 30f

        bitmap = Bitmap.createBitmap(
            windowManager
                .defaultDisplay.width, windowManager
                .defaultDisplay.height, Bitmap.Config.ARGB_8888
        )

        canvas = Canvas(bitmap)
        canvas.drawColor(Color.WHITE) //background
        //
        reusableImageView = findViewById(R.id.ImageViewForDrawing)
        reusableImageView.setImageBitmap(bitmap)
        reusableImageView.visibility = View.VISIBLE

        canvas.drawPoint(1f, 1f, paint)

        spinner.onItemSelectedListener = object : AdapterView.OnItemSelectedListener {
            override fun onItemSelected(
                adapterView: AdapterView<*>?,
                view: View?,
                i: Int,
                l: Long
            ) {
                paint.strokeWidth = dataList[i].toFloat()
            }

            // If no option selected
            override fun onNothingSelected(arg0: AdapterView<*>?) {}
        }

        textView = findViewById(R.id.help)
        textView.text = "Use Arrow Keys to draw ! "
    }

    fun drawright(view: View) {
        reusableImageView.isFocusable = true
        reusableImageView.requestFocus()
        endx += 10
        canvas.drawLine(startx.toFloat(), starty.toFloat(), endx.toFloat(), endy.toFloat(), paint)
        startx = endx
        starty = endy
        //moveRect(canvas);
        reusableImageView.invalidate()
        textView.text = "X= $endx  Y= $endy"
    }

    fun drawdown(view: View) {
        reusableImageView.isFocusable = true
        reusableImageView.requestFocus()
        endy += 10
        canvas.drawLine(startx.toFloat(), starty.toFloat(), endx.toFloat(), endy.toFloat(), paint)
        startx = endx
        starty = endy
        //moveRect(canvas);
        reusableImageView.invalidate()
        textView.text = "X= $endx  Y= $endy"
    }

    fun drawleft(view: View) {
        reusableImageView.isFocusable = true
        reusableImageView.requestFocus()
        endx -= 10
        canvas.drawLine(startx.toFloat(), starty.toFloat(), endx.toFloat(), endy.toFloat(), paint)
        startx = endx
        starty = endy
        //moveRect(canvas);
        reusableImageView.invalidate()
        textView.text = "X= $endx  Y= $endy"
        if (endx<=0)
        {
            val builder = AlertDialog.Builder(this)
            builder.setTitle("Negative X-Axis Value ")
            builder.setMessage("No Line will  be drawn for Negative X-Axis Values!!")
            builder.setPositiveButton("OK", null)
            val dialog = builder.create()
            dialog.show()
        }
    }

    fun drawup(view: View) {
        reusableImageView.isFocusable = true
        reusableImageView.requestFocus()
        endy -= 10
        canvas.drawLine(startx.toFloat(), starty.toFloat(), endx.toFloat(), endy.toFloat(), paint)
        startx = endx
        starty = endy
//moveRect(canvas);
        reusableImageView.invalidate()
        textView.text = "X= $endx Y= $endy"
        if (endy<=0)
        {
            val builder = AlertDialog.Builder(this)
            builder.setTitle("Negative Y-Axis Value ")
            builder.setMessage("No Line will  be drawn for Negative Y-Axis Values!!")
            builder.setPositiveButton("OK", null)
            val dialog = builder.create()
            dialog.show()
        }
    }


    fun redline(view: View) {
        paint.color = Color.RED
    }

    fun cyan(view: View) {
        paint.color = Color.CYAN
    }

    fun yellowline(view: View) {
        paint.color = Color.YELLOW
    }

    fun clearall(view: View) {
        canvas.drawColor(Color.rgb(95, 75, 92)) //background
        canvas.drawColor(Color.WHITE)
        startx = 0
        starty = 0
        endx = 0
        endy = 0
        textView.text = "Use Arrow Keys to draw ! "
    }
}